anesthetic/_matplotlib/core.py

```diff
@@ -0,0 +1,21 @@
+from pandas.plotting._matplotlib.core import ScatterPlot as _ScatterPlot
+from pandas.plotting._matplotlib.core import HexBinPlot as _HexBinPlot
+from anesthetic.weighted_pandas import _WeightedObject
+
+
+class ScatterPlot(_ScatterPlot):
+    # noqa: disable=D101
+    def __init__(self, data, x, y, s=None, c=None, **kwargs) -> None:
+        if isinstance(data, _WeightedObject):
+            data = data.compress()
+            kwargs['alpha'] = kwargs.get('alpha', 0.5)
```

Comment:
New code. Compute weighted scatter plots by compressing to the channel capacity, and switching to alpha of 0.5 (the same convention as `scatter_matrix`), so that repeated/overlapping points come out with greater density.